tools/download.go

```diff
@@ -147,23 +79,15 @@ func pathExists(path string) bool {
 // if it already exists.
 func (t *Tools) Download(pack, name, version, behaviour string) error {
 
-	indexFile := path.Join(t.Directory, "package_index.json")
-	signatureFile := path.Join(t.Directory, "package_index.json.sig")
-
-	if _, err := os.Stat(path.Join(t.Directory, "package_index.json")); err != nil || time.Since(t.LastRefresh) > 1*time.Hour {
+	if !t.Index.IndexFile.Exist() || time.Since(t.Index.LastRefresh) > 1*time.Hour {
 		// Download the file again and save it
-		err = t.DownloadPackageIndex(indexFile, signatureFile)
+		err := t.Index.DownloadAndVerify()
 		if err != nil {
 			return err
 		}
 	}
 
-	err := checkGPGSig(indexFile, signatureFile)
-	if err != nil {
-		return err
-	}
-
-	body, err := os.ReadFile(indexFile)
+	body, err := os.ReadFile(t.Index.IndexFile.String())
```

Comment:
```suggestion
	body, err := t.Index.Read()
```
I'd avoid this code duplication and use a function that encapsulate such logic